### ISSUE TITLE
where >= <= etc.. support?

### ISSUE THREAD

**Issue by DrDeath72** [execut/yii2-1c-odata#1]

**Comment by execut** [comment]
Hello! Yes, 1C OData has support for such functionality, only in a slightly different form. This condition should work ([here docs](https://1c-programmer-blog.ru/programmirovanie/rest-interfejs-odata-v-1c.html)):
```php
$query->andWhere('Цена ge 123'); // "Цена >= 123" 1C oData analog
$query->andWhere('Цена le 123'); // "Цена <= 123" 1C oData analog
```

**Comment by DrDeath72** [comment]
да я так и налепил, но если уж ActiveQuery сделан, то было бы неплохо все что в ней было сделать

а еще orderBy экранирует колонку и не работает, без кавычек работает. с 1с недавно, не знаю косяк это или от версии зависит...
```
// Работает - Period desc
$query->addOrderBy([
	'Period desc',
]);
// Не работает - 'Period' desc
$query->addOrderBy([
	'Period' => SORT_DESC,
]);

```

**Comment by execut** [comment]
Да, я такого с сортировкой не реализовывал, не должно работать. Сделал только то, что нужно было для CRUD-ов 1C-а, построенных с помощью [execut/yii2-crud-fields](https://github.com/execut/yii2-crud-fields). Чтобы добавить поддержку такой сортировки нужно доработать [этот блок](https://github.com/execut/yii2-1c-odata/blob/master/ActiveQuery.php#L197). Я в последние этапы разработки компонента понял, что эффективнее будет перед дальнейшими доработками покрывать тестами и рефакторить. Всё слишком стало хрупко.

Согласен, было-б удобно сделать преобразование стандартных операторов в 1С-овские. Получается вроде этого хотелось бы?
```php
$query->andWhere([
    '>=',
   'Цена',
   100,
]);
```

**Comment by DrDeath72** [comment]
> Чтобы добавить поддержку такой сортировки нужно доработать этот блок. 

Да в том блоке дописывает кавычки

> Получается вроде этого хотелось бы?

Да, стандартные, а то начал писать сравнение дат, цифр и долго втыкал почему не работает

**Comment by execut** [comment]
Тогда запланирую это дело, если руки дойдут до рефакторинга пакета. Но пока нет на это времени, буду рад помощи. Насчёт сортировки добавил отдельную задачу #2

**Comment by DrDeath72** [comment]
сортировка там понимает, просто она излишне экранирует, но это на мой скромный взгляд видящего 1 базу 1с :D
